pkg/sql/sessionprotectedts/session_protected_ts_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sessionprotectedts_test
+
+import (
+	"context"
+	gosql "database/sql"
+	"net/url"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptreconcile"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptstorage"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessionprotectedts"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/uuid"
+	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func createSessionAndProtect(
+	ctx context.Context,
+	t *testing.T,
+	s0 serverutils.ApplicationLayerInterface,
+	sqlDB *sqlutils.SQLRunner,
+	pts protectedts.Storage,
+) (*gosql.DB, *ptpb.Record) {
+	t.Helper()
+
+	getUserConn := func(t *testing.T, server serverutils.ApplicationLayerInterface) *gosql.DB {
+		pgURL := url.URL{
+			Scheme: "postgres",
+			User:   url.UserPassword("foo", "hunter2"),
```

Comment:
:D

I like this. But, if we make the user without a password, you might be able to use SQLConnForUser on the ApplicationLayerInterface for this.

Reply:
good point, we don't even need a user we can just test with root. fixed